packages/flame/lib/src/gestures/events.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:flutter/gestures.dart';
+
+import '../../extensions.dart';
+
+abstract class BaseInfo<T> {
+  final Vector2 position;
+  final T raw;
+
+  BaseInfo(this.position, this.raw);
+}
+
+class TapDownInfo extends BaseInfo<TapDownDetails> {
+  TapDownInfo(Vector2 position, TapDownDetails raw) : super(position, raw);
+}
+
+class TapUpInfo extends BaseInfo<TapUpDetails> {
+  TapUpInfo(Vector2 position, TapUpDetails raw) : super(position, raw);
+}
+
+class LongPressStartInfo extends BaseInfo<LongPressStartDetails> {
+  LongPressStartInfo(
+    Vector2 position,
+    LongPressStartDetails raw,
+  ) : super(position, raw);
+}
+
+class LongPressEndInfo extends BaseInfo<LongPressEndDetails> {
+  final Velocity velocity;
+
+  LongPressEndInfo(
+    Vector2 position,
+    this.velocity,
+    LongPressEndDetails raw,
+  ) : super(position, raw);
+}
+
+class LongPressMoveUpdateInfo extends BaseInfo<LongPressMoveUpdateDetails> {
+  LongPressMoveUpdateInfo(
+    Vector2 position,
+    LongPressMoveUpdateDetails raw,
+  ) : super(position, raw);
+}
+
+class ForcePressInfo extends BaseInfo<ForcePressDetails> {
+  final double pressure;
+
+  ForcePressInfo(
+    Vector2 position,
+    this.pressure,
+    ForcePressDetails raw,
+  ) : super(position, raw);
+}
+
+// TODO(erick) This class, and the next one has a bunch of info, we need to check the ones
+// that could wrapped directly by the info class
+class PointerScrollInfo extends BaseInfo<PointerScrollEvent> {
+  final Vector2 scrollDelta;
+
+  PointerScrollInfo(Vector2 position, this.scrollDelta, PointerScrollEvent raw)
+      : super(position, raw);
+}
+
+class PointerHoverInfo extends BaseInfo<PointerHoverEvent> {
+  PointerHoverInfo(Vector2 position, PointerHoverEvent raw)
+      : super(position, raw);
+}
+
+class DragDownInfo extends BaseInfo<DragDownDetails> {
+  DragDownInfo(Vector2 position, DragDownDetails raw) : super(position, raw);
+}
+
+class DragStartInfo extends BaseInfo<DragStartDetails> {
+  DragStartInfo(Vector2 position, DragStartDetails raw) : super(position, raw);
+}
+
+class DragUpdateInfo extends BaseInfo<DragUpdateDetails> {
+  final Vector2 delta;
```

Comment:
Maybe the creation of these `Vector2` can be "lazy" by using getters somehow? I'm thinking for events that are fired a lot we are going to be creating a lot of objects like this.

Reply:
Not sure I understood, this is created by the flutter callback, so the objects will already be created cannot see how we could make this lazy, maybe I am missing something?

Reply:
This one was a bad example, but it also applies on this one. So on this one we do `toVector2` which creates a new object and this event will trigger a lot so we will create a lot of objects. The same with `game.projectCoordinate`. I'm thinking that the information to create the vectors could exist on the info object so that we don't have to create new objects until they are needed.

Reply:
Got it! Not sure how to implement that, but makes a lot of sense, gonna try think on something

Reply:
I think if either the game or the projection function is passed in you can just make the values getters backed by a nullable value that serves as a memoization of the vector

Reply:
Since this will not change will be an internal thing, I would like to do that in a follow up, as the size of this PR is quite big at this moment. What do you think?

This applies to you suggestion too @renancaraujo